src/applications/TrainingModule.cpp

```diff
@@ -2001,44 +2042,70 @@ std::vector<int> TrainingModule::SVMFFSBasedFeatureSelection(const VariableSizeM
+                result = InternalCrossValidationResubstitution(reducedFeatureSet, labels, pow(2, cValue), pow(2, gValue), classifiertype);
+              else
+                result = InternalCrossValidation(reducedFeatureSet, labels, pow(2, cValue), pow(2, gValue), classifiertype);
+
+              if (result[3] > bestCV)
+                bestCV = result[3];
+            }
+        }
+        else
+        {
+          for (double cValue = -5; cValue <=5; cValue = cValue + 1)
           {
-            VectorDouble result = InternalCrossValidationResubstitution(reducedFeatureSet, labels, pow(2, cValue), pow(2, gValue), classifiertype);
+            VectorDouble result;
+            if (cvtype == 1)
+              result = InternalCrossValidationResubstitution(reducedFeatureSet, labels, pow(2, cValue), 0.01, classifiertype);
+            else
+              result = InternalCrossValidation(reducedFeatureSet, labels, pow(2, cValue), 0.01, classifiertype);
+
             if (result[3] > bestCV)
-            {
-              bestC = pow(2, cValue);
-              bestG = pow(2, gValue);
               bestCV = result[3];
-            }
           }
+        }
       }
       else
       {
-        for (double cValue = -5; cValue <= 5; cValue = cValue + 1)
-        {
-          VectorDouble result = InternalCrossValidationResubstitution(reducedFeatureSet, labels, pow(2, cValue), 0.01, classifiertype);
-          if (result[3] > bestCV)
-          {
-            bestC = pow(2, cValue);
-            bestCV = result[3];
-          }
-        }
+        VectorDouble result;
+        if (cvtype == 1)
+          result = InternalCrossValidationResubstitution(reducedFeatureSet, labels, bestC, bestG, classifiertype);
+        else
+          result = InternalCrossValidation(reducedFeatureSet, labels, bestC, bestG, classifiertype);
+        bestCV = result[3];
       }
-      //VectorDouble result = InternalCrossValidationResubstitution(reducedFeatureSet, traininglabels, 1, 0.1, classifiertype);
-      //bestCV = result[3];
       CrossValidatedBalancedAccuraciesAfterAddingUnselectedFeatures.push_back(bestCV);
     }
     //sort cross-validated balanced accuracies and pick the best selected feature
     int index = std::distance(CrossValidatedBalancedAccuraciesAfterAddingUnselectedFeatures.begin(), std::max_element(CrossValidatedBalancedAccuraciesAfterAddingUnselectedFeatures.begin(), CrossValidatedBalancedAccuraciesAfterAddingUnselectedFeatures.end()));
     CrossValidatedBalancedAccuraciesFinal.push_back(CrossValidatedBalancedAccuraciesAfterAddingUnselectedFeatures[index]);
+    
+    //check if we have three consecutive 1's 
+    int sizeofcvaccuracies = CrossValidatedBalancedAccuraciesFinal.size();
+    if (sizeofcvaccuracies >= 3)
+    {
+      if (CrossValidatedBalancedAccuraciesFinal[sizeofcvaccuracies - 1] == 1 && CrossValidatedBalancedAccuraciesFinal[sizeofcvaccuracies - 2] == 1 && CrossValidatedBalancedAccuraciesFinal[sizeofcvaccuracies - 3] == 1)
```

Comment:
Please add comments about what is happening and what are the hard coded values.

Reply:
already have comments for this segment of code